src/modules/shorthand/mul.rs

```diff
@@ -32,16 +32,17 @@ impl SyntaxModule<ParserMetadata> for ShorthandMul {
     fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
         let var_tok = meta.get_current_token();
         self.var = variable(meta, variable_name_extensions())?;
-        let tok = meta.get_current_token();
         token(meta, "*=")?;
         let variable = handle_variable_reference(meta, var_tok, &self.var)?;
         self.kind = variable.kind;
         self.global_id = variable.global_id;
         self.is_ref = variable.is_ref;
-        self.expr.parse(meta)?;
-        let message = "Multiplication operation can only multiply numbers";
-        let predicate = |kind| matches!(kind, Type::Num);
-        expression_arms_of_type(meta, &self.kind, &self.expr.get_type(), predicate, tok, message)?;
+        syntax(meta, &mut *self.expr)?;
+        if self.kind != self.expr.get_type() || !matches!(self.kind, Type::Num) {
+            let message = format!("Cannot multiply an expression of type '{}' with a variable of type '{}'", self.expr.get_type(), self.kind);
+            let err = self.expr.get_error_message(meta).message(message);
+            return Err(Failure::Loud(err));
+        }
```

Comment:
It would be nice to have macro for this. Could you for example separate the part of `parse_binop` macro that shows these messages to another macro?

Reply:
Uhh sure